staging/src/k8s.io/kubectl/pkg/cmd/wait/wait.go

```diff
@@ -320,6 +328,40 @@ func (o *WaitOptions) RunWait() error {
 	ctx, cancel := watchtools.ContextWithOptionalTimeout(context.Background(), o.Timeout)
 	defer cancel()
 
+	isForDelete := strings.ToLower(o.ForCondition) == "delete"
+	if o.WaitForCreation && o.Timeout == 0 {
+		return fmt.Errorf("--wait-for-creation requires a timeout value greater than 0")
```

Comment:
This is a regressing change for folks using `--timeout=0` as documented:

> The length of time to wait before giving up. Zero means check once and don't wait.

Previously, this would check a condition once:

```go
kubectl wait --for=condition=Available deployments --all --timeout=0
```

Now, it fails with:

```
error: --wait-for-creation requires a timeout value greater than 0
```

I'm not sure defaulting WaitForCreation to true is compatible here. Rollback in https://github.com/kubernetes/kubernetes/pull/125630 with a test added to exercise the `--timeout=0` case so this can be brought back in compatible.

Reply:
Does that mean if the timeout=0, we should ignore the waitforcreation step and mention this in the flag description?

Reply:
Because when timeout=0, `ConditionFn` will be executed once and returns

Reply:
yeah, it should be something like this;

```go
if o.WaitForCreation && !isForDelete && o.Timeout != 0 {
```

Reply:
> Please use the new `--wait-for-creation=false` flag for the previous behavior of immediately failing if the specified resource does not yet exist.

That sounds like this is changing existing behavior, which doesn't really match the description of `This PR introduces new --wait-for-creation flag to also wait for the creation of resources with
preserving the default behavior.`

Would it be simpler and more compatible to default WaitForCreation to false?

Reply:
I agree with you. Besides, we can change the default separately in the future after carefully considering what we should do for such cases.

Reply:
@eddiezane what do you think about bringing same PR back just the default value is false?